lib/router_constraints.rb

```diff
@@ -130,9 +130,18 @@ def matches?(request, path = nil)
 
 class RefIRIRouterConstraint < IRIRouterConstraint
   def matches?(request)
+    regex = %r{
+      (\?[^=]+$) # matches a lone MMT-argument
+      |
+      \?([^=;&]+)[&;].*$ # splits a lone MMT-arg from normal query-string
+      |
+      \?.*$ # drops a the whole standard query-string
+    }x
     # remove the ref/:version_number portion from path
     path = Journey::Router::Utils.unescape_uri(request.original_fullpath).
-      sub(%r{\A/ref/\d+/}, '')
+      sub(%r{\A/ref/\d+/}, '').
+      sub(regex, '\1')
```

Comment:
I'm not sure, to be honest. Access-Tokens are an internal feature (not exposed to the public), and we only supply `/ref/version_number`-based loc/ids to that. So is there any reason to support MMT?